src/openms/source/ANALYSIS/OPENSWATH/SpectraExtractor.cpp

```diff
@@ -0,0 +1,583 @@
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Douglas McCloskey, Pasquale Domenico Colaianni $
+// $Authors: Douglas McCloskey, Pasquale Domenico Colaianni $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/ANALYSIS/OPENSWATH/SpectraExtractor.h>
+
+namespace OpenMS
+{
+  SpectraExtractor::SpectraExtractor() :
+    DefaultParamHandler("SpectraExtractor")
+  {
+    getDefaultParameters(defaults_);
+
+    // write defaults into Param object param_
+    defaultsToParam_();
+  }
+
+  SpectraExtractor::~SpectraExtractor() {}
+
+  void SpectraExtractor::setRTWindow(const double& rt_window)
+  {
+    rt_window_ = rt_window;
+  }
+
+  double SpectraExtractor::getRTWindow() const
+  {
+    return rt_window_;
+  }
+
+  void SpectraExtractor::setMinScore(const double& min_score)
+  {
+    min_score_ = min_score;
+  }
+
+  double SpectraExtractor::getMinScore() const
+  {
+    return min_score_;
+  }
+
+  void SpectraExtractor::setMinForwardMatch(const double& min_forward_match)
+  {
+    min_forward_match_ = min_forward_match;
+  }
+
+  double SpectraExtractor::getMinForwardMatch() const
+  {
+    return min_forward_match_;
+  }
+
+  void SpectraExtractor::setMinReverseMatch(const double& min_reverse_match)
+  {
+    min_reverse_match_ = min_reverse_match;
+  }
+
+  double SpectraExtractor::getMinReverseMatch() const
+  {
+    return min_reverse_match_;
+  }
+
+  void SpectraExtractor::setMZTolerance(const double& mz_tolerance)
+  {
+    mz_tolerance_ = mz_tolerance;
+  }
+
+  double SpectraExtractor::getMZTolerance() const
+  {
+    return mz_tolerance_;
+  }
+
+  void SpectraExtractor::setMZToleranceUnits(const String& mz_tolerance_units)
```

Comment:
using a string might be prone to errors. consider using a bool if you only want to chagne between absolute and relative (ppm) tolerances